src/main/resources/category/java/remoting.xml

```diff
@@ -408,535 +408,584 @@ return HttpClientBuilder.create()
-message="Netflix Hystrix is deprecated. Use an alternative like resilience4j"
-externalInfoUrl="${doc_root}/JavaCodePerformance.md#ibi11">
-<description>Problem: Hystrix is not actively maintained anymore.&#13;
-Solution: Netflix recommends to use open source alternatives like resilience4j. (jpinpoint-rules)</description>
-<priority>3</priority>
-<properties>
-<property name="tags" value="jpinpoint-rule,deprecated" type="String" description="classification"/>
-<property name="xpath">
-    <value><![CDATA[
-//ImportDeclaration/Name[starts-with(@Image, "com.netflix.hystrix")]
-]]>
-    </value>
-</property>
-</properties>
-<example>
-</example>
-</rule>
 
-<rule -name="AvoidRecreatingHttpClient"
-class="net.sourceforge.pmd.lang.rule.xpath.XPathRule"
-language="java"
-message="An HttpClient is created and combined with request-response."
-externalInfoUrl="${doc_root}/JavaCodePerformance.md#ibi12">
-<description>Problem: Apache HttpClient with its connection pool and timeouts should be setup once and then used for many requests. It is quite expensive to create and can only provide the benefits of pooling when reused in all requests for that connection.&#13;
-Solution: Create/build HttpClient with proper connection pooling and timeouts once, and then use it for requests. (jpinpoint-rules)</description>
-<priority>2</priority>
-<properties>
-<property name="tags" value="jpinpoint-rule,performance" type="String" description="classification"/>
-<property name="xpath">
-    <value><![CDATA[
-//TypeDeclaration/ClassOrInterfaceDeclaration[count(.//Annotation//Name[@Image='Configuration']) = 0]
-//MethodDeclaration//VariableInitializer//Name[
-@Image='HttpClientBuilder.create' or @Image='HttpClients.custom' or @Image='HttpAsyncClientBuilder.create']
-[ancestor::MethodDeclaration//ClassOrInterfaceType[@Image='HttpEntity' or @Image='ResponseEntity']]
+    <rule name="AvoidRecreatingHttpClient"
+          class="net.sourceforge.pmd.lang.rule.xpath.XPathRule"
+          language="java"
+          message="An HttpClient is created and combined with request-response."
+          externalInfoUrl="${doc_root}/JavaCodePerformance.md#ibi12">
+        <description>Problem: Apache HttpClient with its connection pool and timeouts should be setup once and then used
+            for many requests. It is quite expensive to create and can only provide the benefits of pooling when reused
+            in all requests for that connection.&#13;
+            Solution: Create/build HttpClient with proper connection pooling and timeouts once, and then use it for
+            requests. (jpinpoint-rules)
+        </description>
+        <priority>2</priority>
+        <properties>
+            <property name="tags" value="jpinpoint-rule,performance" type="String" description="classification"/>
+            <property name="xpath">
+                <value><![CDATA[
+//ClassDeclaration[count(.//Annotation[pmd-java:typeIs('org.springframework.context.annotation.Configuration')]) = 0]
```

Comment:
can use `not(pmd-java:hasAnnotation('...')`?

Reply:
good point, updated